libcxx/docs/Hardening.rst

```diff
@@ -29,8 +29,11 @@ modes are:
   rigour impacts performance more than fast mode: we recommend benchmarking to
   determine if that is acceptable for your program.
 - **Debug mode**, which enables all the available checks in the library,
-  including internal assertions, some of which might be very expensive. This
-  mode is intended to be used for testing, not in production.
+  including heuristic checks that might have significant performance overhead as
+  well as internal library assertions. This mode should be used in
+  non-production environments (such as test suites, CI, local development). We
+  don’t commit to a particular level of performance in this mode and it’s *not*
```

Comment:
I would start the text that it's not intended for production usage, that way the reader can skip this directly when looking for options suitable for production usage.

Reply:
I find it a little hard to rephrase without breaking the flow. I'd prefer mode descriptions to follow the consistent pattern where each paragraph begins with describing the enabled checks, followed by some additional remarks. I think the name `debug` already strongly implies it's not suitable for production use, and the paragraphs are relatively short, so I don't think it should cause any significant confusion to the reader.